src/main/java/io/confluent/kafkarest/SimpleConsumerRecordsCache.java

```diff
@@ -0,0 +1,286 @@
+/**
+ * Copyright 2016 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package io.confluent.kafkarest;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.common.TopicPartition;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NavigableMap;
+import java.util.Queue;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ConcurrentMap;
+
+
+
+public class SimpleConsumerRecordsCache {
```

Comment:
The cache should improve performance in case end user sequentially fetches limited batches of records from a specific partition. The extra records that were fetched are stored within cache. Say topic "events" has 3 partitions. The last record of partition 0 has offset 9999.
The `simpleconsumer.cache.max.records=1000`. User requests `/topic/events/partitions/0/messages&offset=5000?count=1000` and the `KafkaConsumer` fetches internally 5000 records. Then records from offset 6000 to 6999 are stored within the cache. When the user will request records sequentially for the next time we minimize invocation of `KafkaConsumer.poll()`.